### ISSUE TITLE
XDR fails to serialize a transaction that is stored on the ledger

### ISSUE THREAD

**Issue by orbitlens** [stellar/js-stellar-base#558]
**Describe the bug**
I stumbled on the XDR serialization bug during processing of ledger history. During the hash calculation, XDR writer fails to serialize a transaction that has been successfully parsed and stored on the ledger by `stellar-core` in the past.

**What version are you on?**
StellarBase v8.2.1
Node v18.12.0 (also reproduced with v16 and v14)

**To Reproduce**
Copy envelope XDR of this transaction from Horzion:

https://horizon.stellar.org/transactions/cb87c9a4f63231d9b87ace625861a868c760fc7cc58d80f4c4c30766c65c0e72

Code to reproduce an error:

```js
const {TransactionBuilder} = require("stellar-base")
const xdr = 'AAAAA......Yodr1IG'
TransactionBuilder.fromXDR(xdr, 'Public Global Stellar Network ; September 2015').hash()
```

throws an error

```error
RangeError [ERR_OUT_OF_RANGE]: The value of "offset" is out of range. It must be >= 0 and <= 65532. Received 65536
    at new NodeError (node:internal/errors:393:5)
    at boundsError (node:internal/buffer:86:9)
    at checkBounds (node:internal/buffer:55:5)
    at checkInt (node:internal/buffer:74:3)
    at writeU_Int32BE (node:internal/buffer:802:3)
    at Buffer.writeInt32BE (node:internal/buffer:890:10)
    at Cursor.write [as writeInt32BE] ({node_modules_path}\js-xdr@1.3.0\node_modules\js-xdr\lib\cursor.js:180:28)
    at Object.write ({node_modules_path}\js-xdr@1.3.0\node_modules\js-xdr\lib\int.js:31:8)
    at Function.write ({node_modules_path}\js-xdr@1.3.0\node_modules\js-xdr\lib\enum.js:58:16)
    at Function.write ({node_modules_path}\js-xdr@1.3.0\node_modules\js-xdr\lib\union.js:129:22)
```

Other similar transactions:
`5e444cec0f71d455d25ff47ca22c4b6c62f424b07ca31a08593a1dfdf958a791`
`a2522eca6269fc8c6b88b682dfe44a931e2ae0fd39e0616870afcfecf94ea9ed`
`4ffa0023b2c72ca0e24a11ee7875c09789933e667163c1cb83675dbdf76a4e90`

**Expected behavior**
XDR should be able to parse and serialize any transaction which made it into a ledger in the past.

**Comment by overcat** [comment]
Related issue https://github.com/stellar/js-xdr/issues/84

**Comment by Shaptic** [comment]
Fixed in v9.0.0-beta
